controllers/stories.js

```diff
@@ -0,0 +1,80 @@
+const stories = require('../models/stories')
+/**
+ * Creates new story
+ *
+ * @param req {Object} - Express request object
+ * @param req.body {Object} - Story object
+ * @param res {Object} - Express response object
+ */
+const addNewStory = async (req, res) => {
+  try {
+    const body = {
+      ...req.body
+    }
+    const story = await stories.updateStory(body)
+
+    if (story) {
+      return res.json({
+        message: 'Story created successfully!',
+        story: story.storyDetails,
+        id: story.storyId
+      })
+    } else {
+      logger.error('Error while creating new story: Incorrect username passed')
+      return res.boom.badRequest('Unable to add story')
+    }
+  } catch (err) {
+    logger.error(`Error while creating new story: ${err}`)
+    return res.boom.badImplementation('An internal server error occurred')
+  }
+}
+/**
+ * Fetches all the stories
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+const fetchStories = async (req, res) => {
+  try {
+    const allStories = await stories.fetchStories()
+    return res.json({
+      message: 'Stories returned successfully!',
+      stories: allStories.length > 0 ? allStories : []
+    })
+  } catch (err) {
+    logger.error(`Error while fetching stories ${err}`)
+    return res.boom.badImplementation('An internal server error occurred')
+  }
+}
+
+/**
+ * Updates the story
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+const updateStory = async (req, res) => {
+  try {
+    const story = await stories.fetchStory(req.params.id)
+    if (!story.storyData) {
+      return res.boom.notFound('Story not found')
+    }
+
+    const updateStoryResult = await stories.updateStory(req.body, req.params.id)
+    if (updateStoryResult) {
+      return res.status(204).send()
+    } else {
+      logger.error(`Error while updating storyId ${req.params.id}: Incorrect username passed`)
+      return res.boom.badRequest('Unable to update story')
+    }
+  } catch (err) {
+    logger.error(`Error while updating story: ${err}`)
+    return res.boom.badImplementation('An internal server error occurred')
+  }
+}
```

Comment:
Let's either separate Add and Update or rename the function.

Reply:
I am already using `addNewStory` function for Add and `updateStory` for Update. I am not sure what do you mean by separate Add and Update.

Reply:
Oh my bad, I meant for the `updateStory` function in `models/stories` file

Reply:
Ohh okay. I will update the function name in `models/stories.js`